layer/nrlf/nrlf/core/model.py

```diff
@@ -74,27 +76,38 @@ def public_alias(cls) -> str:
         return cls.__name__
 
 
-KEY_SEPARATOR = "#"
-
-
-def key(*args):
+def key(*args) -> str:
     """
-    standard method for creating PKs/SKs
+    Standard method for creating PKs/SKs. Note that the "magic" here
+    is that calling 'str()' on 'DynamoDbStringType' implicitly returns
+    the __root__ value.
     """
-    return KEY_SEPARATOR.join(map(lambda s: str(s), args))
+    return KEY_SEPARATOR.join(map(str, args))
+
+
+def dynamodb_key(*args) -> DynamoDbStringType:
+    return DynamoDbStringType(__root__=key(*args))
+
+
+def convert_document_pointer_id_to_pk(id: str) -> str:
+    producer_id, document_id = _get_tuple_components(id, separator=ID_SEPARATOR)
+    ods_code_parts = producer_id.split(CUSTODIAN_SEPARATOR)
```

Comment:
Is there an assumption that the id provided will always be in the form of `custodian.suffix-document_id`? or does this not throw an error if there wasnt a suffix applicable?

Reply:
It has been revealed to me that if it cannot find the deliminator then it doesnt care 👍